### ISSUE TITLE
Supply precompiled versions of pathwatcher

### ISSUE THREAD

**Issue by jonand** [Galadirith/markdown-preview-plus#81]
I'm attempting to run Atom on a locked down windows 7 computer. Do you really need visual studio to install markdown-preview-plus? That's a rather heavy dependency. I can get e.g. python working with out admin rights, but not visual studio.

~~~
Installing “markdown-preview-plus@1.7.0” failed.Hide output…

> runas@2.0.0 install C:\Users\<user>\AppData\Local\Temp\apm-install-dir-115724-9716-36ftq6\node_modules\markdown-preview-plus\node_modules\pathwatcher\node_modules\runas
> node-gyp rebuild


C:\Users\<user>\AppData\Local\Temp\apm-install-dir-115724-9716-36ftq6\node_modules\markdown-preview-plus\node_modules\pathwatcher\node_modules\runas>node "C:\<user>\Applications\Atom\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild 

> oniguruma@4.2.2 install C:\Users\<user>\AppData\Local\Temp\apm-install-dir-115724-9716-36ftq6\node_modules\markdown-preview-plus\node_modules\highlights\node_modules\first-mate\node_modules\oniguruma
> node-gyp rebuild


C:\Users\<user>\AppData\Local\Temp\apm-install-dir-115724-9716-36ftq6\node_modules\markdown-preview-plus\node_modules\highlights\node_modules\first-mate\node_modules\oniguruma>node "C:\<user>\Applications\Atom\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild 

npm WARN deprecated CSSselect@0.4.1: the module is now available as 'css-select'
npm WARN deprecated CSSwhat@0.4.7: the module is now available as 'css-what'
gypgypnpm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\<user>\\Applications\\Atom\\resources\\app\\apm\\bin\\\\node.exe" "C:\\<user>\\Applications\\Atom\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\<user>\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\<user>\\.atom\\.apmrc" "install" "C:\\Users\\<user>\\AppData\\Local\\Temp\\d-115724-9716-1ku43jk\\package.tgz" "--target=0.22.0" "--arch=ia32"
npm ERR! node v0.10.35
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! runas@2.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the runas@2.0.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the runas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls runas
npm ERR! There is likely additional logging output above.
~~~

**Comment by Galadirith** [comment]
@jonand Unfortunately MPP does depend on several packages that have to be compiled using native build tools, and on Windows that does have to be Visual Studios. I'am sorry that I don't have better news for you on this issue @jonand.

I haven't looked in to this at all so I am only speculating, but it might be possible to distribute compiled distributions of MPP in the future for Windows users, but even if that is something theoretically possible I don't know yet when we could do that.

Thanks @jonand and please post back if you have any more questions :D

**Comment by leipert** [comment]
@jonand @Galadirith 
I will have a look into: http://cylonjs.com/blog/2014/11/19/creating-multiplatform-precompiled-binaries-for-node-modules/
Maybe we can save the compiling for you all.

**Comment by jonand** [comment]
@leipert Precompiled would be nice!

I set up a virtual windows 7 box, installed, compiled, and then copied the markdown-preview-plus package to my locked down work computer. It works, but it won't be much fun to update and I can't really ask my colleagues to do the same!

**Comment by leipert** [comment]
Will look into it this weekend.

**Comment by leipert** [comment]
Unfortunately it is not possible (yet) as `node-pre-gyp` does not supports Atom's shell yet:
https://github.com/mapbox/node-pre-gyp/pull/112

Sorry.

**Comment by Galadirith** [comment]
@leipert I was browsing around and somehow stumbled across dundalek/atom-markdown-mindmap#1. With MPP we're kind of in a fortunate position in that almost all of our dependencies are also dependencies of markdown-preview so I would guess they are part of Atom's core package cache? I will investigate what happens if we remove certain dependencies, and with a bit of luck we might not require any compilation at all.

If this does pan out well we could then demote any such packages to `dev` dependencies.

**Comment by Galadirith** [comment]
With #94 now merged into release `2.1.0` this is no longer nessesary.
